api/applications_api.py

```diff
@@ -0,0 +1,188 @@
+from datetime import datetime
+from dateutil.parser import parse
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from rest_framework import viewsets, serializers
+from rest_framework.response import Response
+from applications.models import (
+    Application,
+    Organisation,
+    ApplicationEvent,
+    Recurrence,
+    Address,
+    Person,
+)
+
+
+MINIMUM_TIME = timezone.datetime(
+    1970, 1, 1, 0, 0, 0, 3, timezone.get_default_timezone()
+)
+
+MAXIMUM_TIME = timezone.datetime(
+    2099, 1, 1, 0, 0, 0, 3, timezone.get_default_timezone()
+)
+
+
+class PersonSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Person
+
+        fields = "__all__"
+
+
+class PersonViewSet(viewsets.ModelViewSet):
+
+    queryset = Person.objects.all()
+
+    serializer_class = PersonSerializer
+
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class AddressSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Address
+
+        fields = "__all__"
+
+
+class AddressViewSet(viewsets.ModelViewSet):
+
+    queryset = Address.objects.all()
+
+    serializer_class = AddressSerializer
+
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class OrganisationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Organisation
+
+        fields = ["id", "name", "identifier", "year_established"]
+
+
+class OrganisationViewSet(viewsets.ModelViewSet):
+
+    queryset = Organisation.objects.all()
+
+    serializer_class = OrganisationSerializer
+
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class ApplicationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Application
+        fields = [
+            "id",
+            "description",
+            "reservation_purpose",
+            "organisation",
+            "contact_person",
+            "user",
+            "organisation",
+        ]
+
+    def validate(self, data):
+        return data
+
+
+class ApplicationViewSet(viewsets.ModelViewSet):
+
+    queryset = Application.objects.all()
+
+    serializer_class = ApplicationSerializer
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+
+class RecurrenceReadSerializer(serializers.ModelSerializer):
```

Comment:
Everything until this point seems relative straightforward, so from here on is stuff i'm not really sure on. If there is a better way to do this, please inform me, i have never worked with django rest framework, so might have missed something very obvious.

Reply:
If we use `RecurrenceField` elsewhere as well, we should consider writing a custom serializer field for it: https://www.django-rest-framework.org/api-guide/fields/#custom-fields instead of handling the field in the serializer that uses it.

Reply:
Ok, i'll take a look into that.